From 0d5bfdca2afa65fc2125766ca6a7643edf0e7d72 Mon Sep 17 00:00:00 2001
From: Vasily Khoruzhick <anarsoul@gmail.com>
Date: Sun, 18 Dec 2011 11:48:46 +0300
Subject: [PATCH 04/12] mtd: nand: s3c2410: add iPAQ specific layout

Signed-off-by: Vasily Khoruzhick <anarsoul@gmail.com>
---
 drivers/mtd/nand/s3c2410.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mtd/nand/s3c2410.c b/drivers/mtd/nand/s3c2410.c
index 4405468..71eeb5b 100644
--- a/drivers/mtd/nand/s3c2410.c
+++ b/drivers/mtd/nand/s3c2410.c
@@ -66,8 +66,8 @@ static const int clock_stop = 0;
 
 static struct nand_ecclayout nand_hw_eccoob = {
 	.eccbytes = 3,
-	.eccpos = {0, 1, 2},
-	.oobfree = {{8, 8}}
+	.eccpos = {8, 9, 10},
+	.oobfree = {{2, 6}, {11, 5}}
 };
 
 /* controller and mtd information */
-- 
1.7.8

