From 00e890820c8d50198da2de86361cc2da7df9b5aa Mon Sep 17 00:00:00 2001
From: Vasily Khoruzhick <anarsoul@gmail.com>
Date: Sat, 17 Dec 2011 13:14:51 +0300
Subject: [PATCH 08/12] ARM: h1940: add bluetooth RF kill switch

Signed-off-by: Vasily Khoruzhick <anarsoul@gmail.com>
---
 arch/arm/mach-s3c2410/mach-h1940.c |   30 ++++++++++++++++++++++++++++++
 1 files changed, 30 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-s3c2410/mach-h1940.c b/arch/arm/mach-s3c2410/mach-h1940.c
index f1764ff..95f231f 100644
--- a/arch/arm/mach-s3c2410/mach-h1940.c
+++ b/arch/arm/mach-s3c2410/mach-h1940.c
@@ -31,6 +31,7 @@
 #include <linux/pda_power.h>
 #include <linux/s3c_adc_battery.h>
 #include <linux/delay.h>
+#include <linux/rfkill-gpio.h>
 
 #include <linux/mtd/mtd.h>
 #include <linux/mtd/partitions.h>
@@ -682,6 +683,21 @@ static struct platform_device h1940_dev_buttons = {
 	}
 };
 
+static struct rfkill_gpio_platform_data h1940_bt_rfkill_pdata = {
+	.name		= "h1940-bt",
+	.reset_gpio	= -EINVAL,
+	.shutdown_gpio	= H1940_LATCH_BLUETOOTH_POWER,
+	.type		= RFKILL_TYPE_BLUETOOTH,
+};
+
+static struct platform_device h1940_bluetooth = {
+	.name		= "rfkill_gpio",
+	.id		= 0,
+	.dev		= {
+		.platform_data = &h1940_bt_rfkill_pdata,
+	},
+};
+
 static struct platform_device *h1940_devices[] __initdata = {
 	&h1940_dev_buttons,
 	&s3c_device_ohci,
@@ -702,6 +718,7 @@ static struct platform_device *h1940_devices[] __initdata = {
 	&s3c_device_ts,
 	&power_supply,
 	&h1940_battery,
+	&h1940_bluetooth,
 };
 
 static void __init h1940_map_io(void)
@@ -776,6 +793,19 @@ static void __init h1940_init(void)
 	gpio_request(H1940_LATCH_SD_POWER, "SD power");
 	gpio_direction_output(H1940_LATCH_SD_POWER, 0);
 
+	/* Configures BT serial port GPIOs */
+	s3c_gpio_cfgpin(S3C2410_GPH(0), S3C2410_GPH0_nCTS0);
+	s3c_gpio_setpull(S3C2410_GPH(0), S3C_GPIO_PULL_NONE);
+	s3c_gpio_cfgpin(S3C2410_GPH(1), S3C2410_GPH1_nRTS0);
+	s3c_gpio_setpull(S3C2410_GPH(1), S3C_GPIO_PULL_NONE);
+	s3c_gpio_cfgpin(S3C2410_GPH(2), S3C2410_GPH2_TXD0);
+	s3c_gpio_setpull(S3C2410_GPH(2), S3C_GPIO_PULL_NONE);
+	s3c_gpio_cfgpin(S3C2410_GPH(3), S3C2410_GPH3_RXD0);
+	s3c_gpio_setpull(S3C2410_GPH(3), S3C_GPIO_PULL_NONE);
+
+	gpio_request(S3C2410_GPC(9), "BT reset");
+	gpio_direction_output(S3C2410_GPC(9), 1);
+
 	platform_add_devices(h1940_devices, ARRAY_SIZE(h1940_devices));
 
 	gpio_request(S3C2410_GPA(1), "Red LED blink");
-- 
1.7.8

