From f89a1a53b00b258aeb44a8511fb4adc657108423 Mon Sep 17 00:00:00 2001
From: Vasily Khoruzhick <anarsoul@gmail.com>
Date: Sun, 18 Dec 2011 11:48:46 +0300
Subject: [PATCH 04/12] mtd: nand: s3c2410: add iPAQ specific layout

Signed-off-by: Vasily Khoruzhick <anarsoul@gmail.com>
---
 drivers/mtd/nand/s3c2410.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mtd/nand/s3c2410.c b/drivers/mtd/nand/s3c2410.c
index d65cbe9..0806e46 100644
--- a/drivers/mtd/nand/s3c2410.c
+++ b/drivers/mtd/nand/s3c2410.c
@@ -54,8 +54,8 @@
 
 static struct nand_ecclayout nand_hw_eccoob = {
 	.eccbytes = 3,
-	.eccpos = {0, 1, 2},
-	.oobfree = {{8, 8}}
+	.eccpos = {8, 9, 10},
+	.oobfree = {{2, 6}, {11, 5}}
 };
 
 /* controller and mtd information */
-- 
1.8.4

