From 7675d11d0adc31ac7b5ad643d72b9ac724b94bdc Mon Sep 17 00:00:00 2001
From: Paul Eggleton <paul.eggleton@linux.intel.com>
Date: Sun, 24 Jul 2011 19:12:32 +0100

---
 config/config.in | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/config/config.in b/config/config.in
index 166f3a9..cf0daa4 100755
--- a/config/config.in
+++ b/config/config.in
@@ -11,11 +11,5 @@ fi
 . @appconfdir@/mach-config-vars
 . @appconfdir@/mach-config-funcs
 
-if [ "x$HAVE_ALSA_MIXER" != "x" -a "x$ALSA_MIXER_INIT" != "x" ]; then
-  if [ ! -e /etc/asound.state ]; then
-    cp $ALSA_MIXER_INIT /etc/asound.state
-    @sbindir@/alsactl restore
-  fi
-fi
 
-export SWITCHEVD_EVENT_SCRIPT=@appconfdir@/scripts/switch-event
\ No newline at end of file
+export SWITCHEVD_EVENT_SCRIPT=@appconfdir@/scripts/switch-event
